msal/src/main/java/com/microsoft/identity/client/MultipleAccountPublicClientApplication.java

```diff
@@ -289,52 +332,116 @@ public void onError(final Exception exception) {
         }
     }
 
+    @Override
+    public IAccount getAccount(@NonNull String identifier) throws InterruptedException, MsalException {
+
+        throwOnMainThread("getAccount");
+
+        final ResultFuture<AsyncResult<IAccount>> future = new ResultFuture<>();
+
+        getAccount(identifier, new GetAccountCallback() {
+            @Override
+            public void onTaskCompleted(IAccount result) {
+                future.setResult(new AsyncResult<IAccount>(result, null));
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                future.setResult(new AsyncResult<IAccount>(null, exception));
+            }
+        });
+
+        AsyncResult<IAccount> result = future.get();
+
+        if(result.getSuccess()){
+            return result.getResult();
+        }else{
+            throw result.getException();
+        }
+
+    }
+
     @Override
     public void removeAccount(@Nullable final IAccount account,
-                              @NonNull final TaskCompletedCallbackWithError<Boolean, Exception> callback) {
+                              @NonNull final RemoveAccountCallback callback) {
         ApiDispatcher.initializeDiagnosticContext();
 
         // First, cast the input IAccount to a MultiTenantAccount
         final MultiTenantAccount multiTenantAccount = (MultiTenantAccount) account;
 
         //create the parameter
+        if (null == multiTenantAccount) {
+            com.microsoft.identity.common.internal.logging.Logger.warn(
+                    TAG,
+                    "Requisite IAccount or IAccount fields were null. Insufficient criteria to remove IAccount."
+            );
+
+            callback.onError(new MsalClientException(MsalClientException.INVALID_PARAMETER));
+            return;
+        }
+
+        final OperationParameters params = OperationParametersAdapter.createOperationParameters(mPublicClientConfiguration);
+
+        // TODO Clean this up, only the cache should make these records...
+        // The broker strips these properties out of this object to hit the cache
+        // Refactor this out...
+        final AccountRecord requestAccountRecord = new AccountRecord();
```

Comment:
@iambmelt do we have a function to extract AccountRecord from a MultiTenantAccount?